src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -132,13 +132,47 @@ public void submit(TimerTask timerTask, long millisecondsDelay) {
         timer.schedule(timerTask, millisecondsDelay);
     }
 
+    /**
+     * Shuts everything down. Upon termination, this method returns.
+     */
     public void shutdownEverything() {
-        // those threads will be allowed to finish
-        this.executorService.shutdown();
-        // those threads will be interrupted in their current task
-        this.lowPriorityExecutorService.shutdownNow();
         // kill the remote thread
         stopRemoteThread();
+
+        try {
```

Comment:
extract try block to method and call it for executorService and lowPriorityExecutorService